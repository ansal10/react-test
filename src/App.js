import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';
import axios from 'axios';


class App extends Component {

	state = {
		firstButtonValue: 'Not Clicked',
		secondButtonValue: 'Not Clicked',
		thirdButtonValue: 'Not Clicked'
	};

	componentDidMount = () => {

	};

	firstButtonClick = (e) => {
		this.setState({
			firstButtonValue: 'Clicked'
		})
	};

	secondButtonClick = (e) => {
		this.setState({
			secondButtonValue: 'Clicked'
		})
	};

	thirdButtonClick = (e) => {
		this.setState({
			thirdButtonValue: 'Clicked'
		})
	};



	render() {
		return (
			<div className="App">
				<button className="firstButton" onClick={this.firstButtonClick}>
					First Buttons
				</button>
				<p>{this.state.firstButtonValue}</p>
				<button className="secondButton" onClick={this.secondButtonClick}>
					Second Button
				</button>
				<p>{this.state.secondButtonValue}</p>
				<button className="thirdButton" onClick={this.thirdButtonClick}>
					Third Button
				</button>
				<p>{this.state.thirdButtonValue}</p>

			</div>
		);
	}
}

export default App;
