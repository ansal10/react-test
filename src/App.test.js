import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Enzyme, {shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });

describe('App should render', () => {

	it('renders without crashing', () => {
		const wrapper = shallow(<App/>);
		const buttons = wrapper.state();
		expect(Object.keys(buttons).length).toEqual(3);
	});

	it('should render 3 P Comp', function () {
		const wrapper = shallow(<App/>);
		const pElements = wrapper.find('p');
		expect(pElements.length).toEqual(3)
		const buttonElements = wrapper.find('button');
		expect(buttonElements.length).toEqual(3)
	});

	it('should match the snapshot', function () {
		const jsonTree = renderer.create(<App/>);
		expect(jsonTree).toMatchSnapshot();
	});

	it('should match the snapshot after first button clicked', function () {
		const jsonTree = renderer.create(<App/>);
		expect(jsonTree).toMatchSnapshot();
	});

})
