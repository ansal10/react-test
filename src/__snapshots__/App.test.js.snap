// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render should match the snapshot 1`] = `
<div
  className="App"
>
  <button
    className="firstButton"
    onClick={[Function]}
  >
    First Buttons
  </button>
  <p>
    Not Clicked
  </p>
  <button
    className="secondButton"
    onClick={[Function]}
  >
    Second Button
  </button>
  <p>
    Not Clicked
  </p>
  <button
    className="thirdButton"
    onClick={[Function]}
  >
    Third Button
  </button>
  <p>
    Not Clicked
  </p>
</div>
`;
